pkg/osbuild/anaconda_stage.go

```diff
@@ -30,10 +30,13 @@ func defaultModuleStates() map[string]bool {
 	}
 }
 
-func enableModules(states map[string]bool, additional []string) {
-	for _, modname := range additional {
+func setModuleStates(states map[string]bool, enable, disable []string) {
```

Comment:
Should we check that `enable` and `disable` are disjoint?

Reply:
It crossed my mind and I'm not sure.  I thought about this a bit and landed on allowing a module to appear in both lists, making it possible to disable a module that's in the enabled list (the tests even cover this case explicitly).  My reasoning is that in image type defaults, we only ever **enable** modules.  The **disable** list comes straight from the user blueprint.  So by letting a module appear in both lists, we essentially give the user the ability to **disable** a module that is **enabled** by default.

For example, if a user is building an image which **enables** the Users module by default and they want to **disable** it for some reason (automated installation, maybe they want to create a user with initial-setup, etc), or if they want to **disable** the Storage module (which we always enable) to do partitioning in a kickstart `%pre` without having Anaconda block an automated installation, this gives them that control.

Reply:
One thing we can do to make things a bit saner is to raise an error when the user adds the same module to both lists in the blueprint itself.  That would make things less ambiguous for the user (they can't enable and disable the same module in the same build) while also letting them disable a default module.

Reply:
So, I've added the question while going through the beginning of the patch set and ended up with pretty much the same conclusion as you, by the end of it. But I thought it would make sense to have this discussion.

WRT the BP customization. I think that either what you suggested would be good, or explicitly documenting the behavior of it (that disabling a module has precedence).

Reply:
Prepared docs PR: https://github.com/osbuild/osbuild.github.io/pull/100